tools/lsp/server_loop.rs

```diff
@@ -109,6 +109,12 @@ impl DocumentCache {
     }
 }
 
+impl crate::properties::OffsetToPositionMapper for (&mut DocumentCache, lsp_types::Url) {
```

Comment:
Instead of a trait, why not simply a struct:
```
struct OffsetToPositionMapper(&mut DocumentCache, lsp_types::Url);
```

It a trait not over-engineered?